design.md

```diff
@@ -0,0 +1,226 @@
+# Design proposal
+
+## Objective
+
+Implement a simple remote process supervisor that lets its users start, stop and monitor arbitrary commands, as well as retrieving their outputs as streams. A CLI tool will be provided to interact with the server via its gRPC-powered API.
+
+## Assumptions, decisions and tradeoffs
+- Given many different implementation options, the most straighforward one will be chosen unless further requirements are provided
+- The jobs provided by users are well-intentioned and not malicious, the resource control mechanisms described below act as a safeguard against user/software errors, not targeted attacks
+- There will not be any attempts to persist the jobs or recover them on failure
+- The output from the jobs will be held in memory until consumed or discarded
```

Comment:
> held in memory until consumed

Could you clarify the "until consumed" part please? Does this mean only 1 client will be able to read the output after which it will be discarded?

Reply:
Yes, that's what I meant, I'm not proposing any kind of internal buffering (to keep things simple), but please let me know if I should do so.

Reply:
Please disregard my last comment as I will change the design now that I understand better the requirements.